pandas/_libs/groupby.pyx

```diff
@@ -556,6 +590,10 @@ def group_add(add_t[:, ::1] out,
                 for j in range(K):
                     val = values[i, j]
 
+                    #  function to release gil for the whole block
+                    if (val == MAXfloat64) or (val == MINfloat64):
+                        sumx[lab, j] = val
+                        break
```

Comment:
If we just break here, does this change behavior for a case like `np.inf + -np.inf`, which should be `np.nan`?

Reply:
yes you're right, this only uses the last occurance of `np.nan/-np.nan` for all following elements. Correctly it should be `np.nan`. Am going to fix this.

Reply:
is fixed, also added a test for this case